soc/nordic/common/pm_s2ram.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file Common pm_s2ram.h include for Nordic SoCs.
+ */
+
+#ifndef _ZEPHYR_SOC_ARM_NORDIC_NRF_PM_S2RAM_H_
+#define _ZEPHYR_SOC_ARM_NORDIC_NRF_PM_S2RAM_H_
+
+/**
+ * @brief Save CPU state on suspend
+ *
+ * This function is used on suspend-to-RAM (S2RAM) to save the CPU state in
+ * (retained) RAM before powering the system off using the provided function.
+ * This function is usually called from the PM subsystem / hooks.
+ *
+ * The CPU state consist of internal registers and peripherals like
+ * interrupt controller, memory controllers, etc.
+ *
+ * @param system_off	Function to power off the system.
+ *
+ * @retval 0		The CPU context was successfully saved and restored.
+ * @retval -EBUSY	The system is busy and cannot be suspended at this time.
+ * @retval -errno	Negative errno code in case of failure.
+ */
+int pm_s2ram_suspend(pm_s2ram_system_off_fn_t system_off);
```

Comment:
why are you exposing this function in a header? sounds like a copy of `arch_pm_s2ram_suspend`?

Reply:
not a copy, rather an extension:
https://github.com/zephyrproject-rtos/zephyr/pull/73095/files#diff-05bd2cd035d6ae43221ff42f2e8e898f073a11f9e1f85ba85f1e10f27463cba8R112-R129
Should it ba called another way?